CoreSecurity/test/unit/src/au/gov/asd/tac/constellation/security/password/PasswordUtilitiesNGTest.java

```diff
@@ -0,0 +1,124 @@
+ */
+public class PasswordUtilitiesNGTest {
+
+    /**
+     * Test of getIV method, of class PasswordUtilities.
+     */
+    @Test
+    public void testGetIV() {
+        System.out.println("getIV");
+        PasswordUtilities.reset();
+
+        final byte[] firstResult = PasswordUtilities.getIV();
+        assertNotNull(firstResult);
+        final byte[] secondResult = PasswordUtilities.getIV();
+        assertEquals(firstResult, secondResult);
+    }
+
+    /**
+     * Test of getIV method, of class PasswordUtilities.
+     */
+    @Test
+    public void testGetIVException() {
+        System.out.println("getIV Exception");
+        PasswordUtilities.reset();
+
+        // Exception can only be tested first
+        final Lookup mockLookup = mock(Lookup.class);
+        when(mockLookup.lookup(PasswordSecret.class)).thenReturn(null);
+        try (final MockedStatic<Lookup> mockLookupStatic = Mockito.mockStatic(Lookup.class)) {
+            // Setup static mock
+            mockLookupStatic.when(Lookup::getDefault).thenReturn(mockLookup);
+
+            PasswordUtilities.getIV();
+            // If exception was not thrown, fail
+            fail("NoSuchElementException should have been thrown");
+        } catch (NoSuchElementException e) {
+            // Left blank
+        }
+    }
+
+    /**
+     * Test of getKey method, of class PasswordUtilities.
+     */
+    @Test
+    public void testGetKey() {
+        System.out.println("getKey");
+        PasswordUtilities.reset();
+
+        final byte[] firstResult = PasswordUtilities.getKey();
+        assertNotNull(firstResult);
+        final byte[] secondResult = PasswordUtilities.getKey();
+        assertEquals(firstResult, secondResult);
+    }
+
+    /**
+     * Test of getKey method, of class PasswordUtilities.
+     */
+    @Test
+    public void testGetKeyException() {
+        System.out.println("getKey Exception");
+        PasswordUtilities.reset();
+
+        // Exception can only be tested first
+        final Lookup mockLookup = mock(Lookup.class);
+        when(mockLookup.lookup(PasswordSecret.class)).thenReturn(null);
+        try (final MockedStatic<Lookup> mockLookupStatic = Mockito.mockStatic(Lookup.class)) {
+            // Setup static mock
+            mockLookupStatic.when(Lookup::getDefault).thenReturn(mockLookup);
+
+            PasswordUtilities.getKey();
+            // If exception was not thrown, fail
+            fail("NoSuchElementException should have been thrown");
+        } catch (NoSuchElementException e) {
+            // Left blank
+        }
+    }
+
+    /**
+     * Test of generateKey method, of class PasswordUtilities.
+     */
+    @Test
+    public void testGenerateKey() {
+        System.out.println("generateKey");
+        try {
+            final byte[] result = PasswordUtilities.generateKey();
+            assertNotNull(result);
+        } catch (NoSuchAlgorithmException e) {
+            fail("NoSuchAlgorithmException exception");
+        }
```

Comment:
this could be changed to a throw instead

Reply:
Rather than catching and re-throwing the same exception. I've removed the try catch block and added `throws` to the test function itself. Do you think this would cause any issues?

Reply:
My apologies, when I meant change to a throw, I meant what you ended up doing (i.e. throw the exception rather than catch it) as appose to throwing an exception in the catch block. It shouldn't cause any unintended issues, as the test will just fail if it encounters an exception. The only reason I would use a catch block in a test is if you do something with the test afterwards that is more than just failing